pinc/ProofProject.inc

```diff
@@ -0,0 +1,170 @@
+<?php
+declare(strict_types=1);
+
+class ProofProjectPage extends LPage
+{
+    public function __construct(ProofProject $proof_project, ProjectPage $project_page)
+    {
+        parent::__construct($proof_project->project, $project_page->page_name, $project_page->page_state, 0);
+    }
+
+    public function pp_checkout($user): array
```

Comment:
Any reason this and `pp_resume_page()` have a `pp_` prefix and the other actions don't? the `pp_` seems superfluous.

Reply:
I originally didn't have these prefixes but at one point I must have added type declarations to the corresponding functions in LPage (returning void). Then phpunit complains that the functions must be compatible with the parent functions so I changed the names. The type declarations in LPage go removed again at some point. Perhaps I should put them back in.

Reply:
Ah, I get it. Let's leave it as-is for now then.

Reply:
Sorry to be so late to this conversation. I don't feel strongly about changing this for this PR, but I'd _**really**_ like to see it changed at some point. Elsewhere in the code, `pp_` indicates post-processing. If a prefix is necessary, could it be changed to something like `pr_`, `pf_` or `ppp_`?

Reply:
Good point, this pp_was based on the pp in PPage (presentation page) but someting else would be better.

Reply:
The reason I'm not _too_ worried about this at this time is because eventually we'll be removing the non-API uses of the PPage code and we can update the definitions in PPage to match what we need for the API and remove the `pp_` prefix.